src/client/common/installer/productInstaller.ts

```diff
@@ -126,6 +126,54 @@ export abstract class BaseInstaller {
             .install(product, resource, cancel, reInstallAndUpdate);
     }
 
+    public async getVersion(product: Product, resource?: InterpreterUri): Promise<string | undefined> {
+        // User may have customized the module name or provided the fully qualified path.
+        const interpreter = isResource(resource) ? undefined : resource;
+        const uri = isResource(resource) ? resource : undefined;
+        const executableName = this.getExecutableNameFromSettings(product, uri);
+
+        const pythonProcess = await this.serviceContainer
+            .get<IPythonExecutionFactory>(IPythonExecutionFactory)
+            .createActivatedEnvironment({ resource: uri, interpreter, allowEnvironmentFetchExceptions: true });
+
+        const output = pythonProcess.execObservable(['-m', 'pip', 'show', executableName], {});
```

Comment:
Does this work for conda? Meaning will running pip show give you the correct information in a conda environment?

Reply:
Why not use `-m ipython --version` or the existing api as mentioned by ian

Reply:
This is the reason:
![image](https://user-images.githubusercontent.com/20347396/123492534-9a155280-d5ce-11eb-9c24-ad20b018d102.png)

I think the python api uses the bottom command, and it gives the version of something different (its ipython) and that wouldn't work. I'll check their code, but for this PR we might want to do it ourselves. 

And fix it for Conda of course.

Reply:
I'm not a python expert (could consider asking the python folks). But an easier way if we can't use the python -m version check or python api then might be to activate the environment and do like this:

python -c "import ipykernel; print(ipykernel.__version__)"

Reply:
I'll try it, although it would be nice to not have to activate it